### DISCUSSION TITLE
v1.11

### DISCUSSION THREAD

**Discussion by dioannidis** [dioannidis/usbasp#27] [Announcements]
1. Differentiate multiple USBasp programmers using Serial Number
2. Add support for AT89S51/2 ICs

<hr /><em>This discussion was created from the release <a href='https://github.com/dioannidis/usbasp/releases/tag/v1.11'>v1.11</a>.</em>

**Comment by afiglee** [comment]
Cool! Will try it

**Comment by harryb777** [comment]


**Reply by dioannidis** [reply]
@harryb777 

> I followed all the instructions provided in the README file. However, when attempting to flash the firmware using AVRDUDESS, I receive the following error: Unknown signature 1e9307 (probably m8)

I never used AVRDUDESS but did you try to select the atmega8 mcu in the MCU (-p) dropbox ( top right as i see your screenshot ) ?

After all, AFAIU, AVRDUDESS calls avrdude underneath ( from the windows title I assume that it calls avrdude 7.2 ) and if the command line options which use to call avrdude are those that are shown ( -c usbasp -B 0.5 ) then it's missing the -p parameter which is the part’s id .

See [2.1 Option Descriptions](https://avrdudes.github.io/avrdude/8.0/avrdude_4.html#Option-Descriptions)

**Reply by harryb777** [reply]
@dioannidis Thank you for your quick reply.

I tried your suggestion, but Dropbox only shows the device if I press the Detect button. If the device is successfully detected, it shows the detected device name. However, the -p command only works after the device has been detected.

Is there any other way I can load the .hex file without using AVRDUDESS?

**Reply by dioannidis** [reply]
@harryb777 

> Is there any other way I can load the .hex file without using AVRDUDESS?

Use avrdude and command line . 

Download avrdude 7.3 from [AVRDUDE v7.3](https://github.com/mcuee/avrdude/releases/tag/v7.3) . If you gonna use a USBasp device with this firmware and you're on Windows use the avrdude-v7.3-windows_mingw-x64.zip archive .

Then use the following command ( I assume that you've have two USBasp devices . One to use as a programmer and the other as the target to be programmed ):

```
avrdude -c usbasp -p atmega8 -U flash:w:USBasp_ATMega8_12MHz_TPI_HIDUART_SNWRITE.hex:i -U eeprom:w:USBasp_EEPROM.eep.hex:i -v
```

**Reply by harryb777** [reply]
@dioannidis Thank you.

It's now working perfectly without any errors — I’ve tested it successfully.

>Then use the following command (I assume that you have two USBasp devices: one as a programmer and the other as the target)

Yes, that's correct — I’m using two USBasp devices: one as the programmer and the other as the target. 

Thanks again for your help!

**Reply by dioannidis** [reply]
@harryb777 

  I suggest that you should replace the avrdudess's avrdude.exe and avrdude.conf  with the newer avrdude ~~7.3~~ 8.0 ( with a MINGW build if you use this USBasp's firmware ) .

**Reply by dioannidis** [reply]
@harryb777 

Also there was a bug in avrdudess, detecting atmega8 which is fixed in the next avrdudess's release. See https://github.com/ZakKemble/AVRDUDESS/commit/b3754dca93ef27aab2023785f8d8cdde88167c45

**Reply by harryb777** [reply]
@dioannidis I did it - the device is detected and the signature is displayed, but the device name doesn’t show in the MCU (-p). Also, the 'Program' button stays disabled even after detection, as shown in the photo.
![image](https://github.com/user-attachments/assets/0405cef0-227e-4cc3-adbd-b1f5b76f2e58)

**Reply by harryb777** [reply]
> Also there was a bug in avrdudess, detecting atmega8 which is fixed in the next avrdudess's release. See [ZakKemble/AVRDUDESS@b3754dc](https://github.com/ZakKemble/AVRDUDESS/commit/b3754dca93ef27aab2023785f8d8cdde88167c45)

Genau, It works with the command line, but it's not working with AVRDudeGUI. That's why I also believe the issue might be with AVRDudeGUI itself.

Thank you very much for point out the issue.

**Reply by dioannidis** [reply]
@harryb777 

I can't help you much with avrdudess . Maybe an upgrade to the latest [avrdudess](https://github.com/ZakKemble/AVRDUDESS/releases) version will help ? You have 2.4, the latest is 2.18 ...

Also by mistake ( copy & paste ) I told you to use avrdude 7.3 . The latest and greatest is [avrdude 8.0](https://github.com/avrdudes/avrdude/releases/tag/v8.0) . Please use that version. And read the Release notes !

**Reply by harryb777** [reply]
@dioannidis I have modified avrdudess code now it's working perfectly. Thank you.
